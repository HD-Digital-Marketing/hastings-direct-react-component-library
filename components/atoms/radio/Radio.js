import React from 'react'
import classNames from 'classnames'
import PropTypes from 'prop-types'
import './radio.scss'

const Radio = props => {

    const {
        className = '',
        theme = '',
        name,
        id,
        value,
        checked = false,
        disabled = false,
        readOnly = false,
        children
    } = props

    const radioClass = classNames('a-radio', className, {
        'a-radio--on-dark': theme === 'dark'
    })

    return (
        <div className={radioClass}>
            <input type="radio" {...{id, name, value, disabled, readOnly}} defaultChecked={checked} />
            <label htmlFor={name}>
                {children}
            </label>
        </div>
    )
}

Radio.propTypes = {
    type: PropTypes.string,
    className: PropTypes.string,
    theme: PropTypes.string,
    name: PropTypes.string,
    id: PropTypes.string,
    value: PropTypes.string,
    checked: PropTypes.bool,
    disabled: PropTypes.bool,
    readOnly: PropTypes.bool,
    children: PropTypes.string,
}

export default Radio
