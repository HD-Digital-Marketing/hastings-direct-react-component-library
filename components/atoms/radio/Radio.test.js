import React from 'react';
import renderer from 'react-test-renderer';
import Radio from './Radio';

describe('Check the radio renders', () => {
    test('it should match the snapshot', () => {

        const tree = renderer
            .create( <Radio id="checkboxOne" name="checkboxOne" checked={true}>Some value to choose</Radio>)
            .toJSON();
        expect(tree).toMatchSnapshot();

    });
});