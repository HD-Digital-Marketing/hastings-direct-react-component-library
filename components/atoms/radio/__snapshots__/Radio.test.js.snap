// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check the radio renders it should match the snapshot 1`] = `
<div
  className="a-radio"
>
  <input
    defaultChecked={true}
    disabled={false}
    id="checkboxOne"
    name="checkboxOne"
    readOnly={false}
    type="radio"
  />
  <label
    htmlFor="checkboxOne"
  >
    Some value to choose
  </label>
</div>
`;
